camayoc/tests/qpc/cli/test_sources.py

```diff
@@ -1030,6 +1030,471 @@ def test_edit_port_negative(
+    """Try to edit a source with an invalid ssl-protocol.
+
+    :id: 3cce79a4-9ac2-41b9-b0eb-5e75ef6264d0
+    :description: Try to edit the ssl-protocol of a source entry. Check that
+        network source entries don't accept that option and the host manager
+        source entries don't accept invalid values
+    :steps: Run ``qpc source edit --name <name> --ssl-protocol
+        <invalid-ssl-protocol>``
+    :expectedresults: The source must not be updated.
+    """
+    cred_name = utils.uuid4()
+    hosts = '127.0.0.1'
+    name = utils.uuid4()
+    port = default_port_for_source(source_type)
+    show_output_dict = {
+        'cred_name': cred_name,
+        'hosts': hosts,
+        'name': name,
+        'port': port,
+        'source_type': source_type,
+    }
+    if source_type == 'network':
+        add_command = (
+            'qpc source add --name {} --cred {} --hosts {} --port {} --type {}'
+            .format(name, cred_name, hosts, port, source_type)
+        )
+        new_ssl_protocol = random.choice(QPC_SSL_PROTOCOL_VALUES)
+        expected_error = (
+            'Error: Invalid SSL options for network source: ssl_protocol')
+        exitstatus = 1
+    else:
+        ssl_protocol = random.choice(QPC_SSL_PROTOCOL_VALUES)
+        add_command = (
+            'qpc source add --name {} --cred {} --hosts {} --port {} '
+            '--ssl-protocol {} --type {}'
+            .format(name, cred_name, hosts, port, ssl_protocol, source_type)
+        )
+        new_ssl_protocol = utils.uuid4()
+        expected_error = (
+            'qpc source edit: error: argument --ssl-protocol: invalid '
+            'choice: \'{}\' \\(choose from \'SSLv23\', \'TLSv1\', '
+            '\'TLSv1_1\', \'TLSv1_2\'\\)'
+            .format(new_ssl_protocol)
+        )
+        exitstatus = 2
+        show_output_dict['options'] = {
+            'ssl_protocol': ssl_protocol,
+        }
+    cred_add_and_check(
+        {
+            'name': cred_name,
+            'username': utils.uuid4(),
+            'password': None,
+            'type': source_type,
+        },
+        [(CONNECTION_PASSWORD_INPUT, utils.uuid4())],
+    )
+
+    qpc_source_add = pexpect.spawn(add_command)
+    assert qpc_source_add.expect(pexpect.EOF) == 0
+    qpc_source_add.close()
+    assert qpc_source_add.exitstatus == 0
+
+    qpc_source_edit = pexpect.spawn(
+        'qpc source edit --name {} --ssl-protocol {}'
+        .format(name, new_ssl_protocol)
+    )
+    assert qpc_source_edit.expect(expected_error) == 0
+    assert qpc_source_edit.expect(pexpect.EOF) == 0
+    qpc_source_edit.close()
+    assert qpc_source_edit.exitstatus == exitstatus
+
+    source_show_and_check(
+        {'name': name},
+        generate_show_output(show_output_dict)
+    )
+
+
+@pytest.mark.parametrize('source_type', QPC_HOST_MANAGER_TYPES)
+def test_edit_disable_ssl(
+        isolated_filesystem, qpc_server_config, source_type):
+    """Edit a source's disable-ssl.
```

Comment:
again small nitpick I think each of these should be `Edit a source's <name of flag> settings.` or alternatively `options`